core/controllers/access_validators_test.py

```diff
@@ -563,3 +568,108 @@ def test_validation_returns_false_if_given_user_is_non_existent(
+            'acquired_skill_ids': ['skill_id_1', 'skill_id_2'],
+            'prerequisite_skill_ids': [],
+            'outline': '',
+            'outline_is_finalized': False,
+            'exploration_id': self.exp_id,
+            'status': 'Draft',
+            'planned_publication_date_msecs': 100,
+            'last_modified_msecs': 100,
+            'first_publication_date_msecs': None,
+            'unpublishing_reason': None
+        }
+
+        self.save_new_skill('skill_id_1', self.admin_id, description='Skill 1')
+        self.save_new_skill('skill_id_2', self.admin_id, description='Skill 2')
+
+        self.story = story_domain.Story.create_default_story(
+            self.story_id_1, 'Public Story Title', 'Description', self.topic_id,
+            self.story_url_fragment_1)
+        self.story.story_contents.nodes = [
+            story_domain.StoryNode.from_dict(self.node_1)
+        ]
+        self.story.story_contents.initial_node_id = self.node_id
+        self.story.story_contents.next_node_id = self.node_id_2
+        story_services.save_new_story(self.admin_id, self.story)
+
+        self.story_2 = story_domain.Story.create_default_story(
+            self.story_id_2, 'Private Story Title', 'Description',
+            self.topic_id, self.story_url_fragment_2)
+        story_services.save_new_story(self.admin_id, self.story_2)
+        subtopic_1 = topic_domain.Subtopic.create_default_subtopic(
+            1, 'Subtopic Title 1', 'url-frag-one')
+        subtopic_1.skill_ids = ['skill_id_1']
+        subtopic_1.url_fragment = 'sub-one-frag'
+        self.save_new_topic(
+            self.topic_id, 'user', name='Topic',
+            description='A new topic',
+            canonical_story_ids=[self.story_id_1, self.story_id_3],
+            additional_story_ids=[], uncategorized_skill_ids=[],
+            subtopics=[subtopic_1], next_subtopic_id=2)
+        topic_services.publish_topic(self.topic_id, self.admin_id)
+        topic_services.publish_story(
+            self.topic_id, self.story_id_1, self.admin_id)
+
+        self.login(self.VIEWER_EMAIL)
+
+    def test_any_user_can_access_review_tests_page(self) -> None:
+        self.get_html_response(
+            '%s/can_access_review_tests_page/staging/topic/%s'
+            % (ACCESS_VALIDATION_HANDLER_PREFIX, self.story_url_fragment_1),
+            expected_status_int=200)
+
+    def test_no_user_can_access_unpublished_story_review_sessions_page(
+        self
+    ) -> None:
+        self.get_html_response(
+            '%s/can_access_review_tests_page/staging/topic/%s'
+            % (ACCESS_VALIDATION_HANDLER_PREFIX, self.story_url_fragment_2),
+            expected_status_int=404)
+
+    def test_get_fails_when_story_doesnt_exist(self) -> None:
+        self.get_html_response(
+            '%s/can_access_review_tests_page/staging/topic/%s'
+            % (ACCESS_VALIDATION_HANDLER_PREFIX, 'non-existent-story'),
+            expected_status_int=302)
```

Comment:
Hello @seanlip those tests were not mine I copied them while migrating but I think you are right we should change the test name 
Below you will find the tests I was coping (the backend handler tests)
https://github.com/oppia/oppia/blob/3c3e5fafebede8cd82b54882ca63e12c13094ddf/core/controllers/review_tests_test.py#L129-L133

Reply:
Yup, some old code may not be 100% correct. Let's take the opportunity to fix it if we change it, otherwise this problem will persist for future devs. Thanks for checking!

Reply:
I have fixed the errors by changing the get_html_response to a get_json and changing the expected code to be 404. (since the redirections are now handled in the frontend the test won't return a 302 error it will just return the 404)